feat: implement allowlist and blocklist apis

akhileshthite opened the pull request:
https://github.com/hyphacoop/distributed-press-organizing/issues/89

RangerMauve commented on api/index.ts:
```diff
@@ -117,18 +117,44 @@ const v1Routes = (cfg: APIConfig, store: Store) => async (server: FastifyTypebox
-  server.get('/blocklist', async (request, reply) => {})
+  server.get('/blocklist', async (request, reply) => {
+    const blockedAccounts = await store.blocklist.list();
+    reply.send(blockedAccounts.join('\n'));
```
We should set th3e content type to text/plain or similar if it isn't already

RangerMauve replied:
Also could you add types to the route the way I have them in the inbox? (same ofr the other endpoints)

RangerMauve commented on api/index.ts:
```diff
@@ -214,29 +240,61 @@ const v1Routes = (cfg: APIConfig, store: Store) => async (server: FastifyTypebox
   // The ID is the URL encoded id from the inbox activity
   server.delete('/:domain/inbox/:id', async (request, reply) => {
-    const { domain, id } = request.params
+    const { domain, id } = request.params as { domain: string, id: string }
```
After setting the types in the request the `as` should be removable

RangerMauve reviewed (changes requested):
Great start! Added some comments, we'll also need the new class for checking if an account is blocked.

akhileshthite commented:
> Great start! Added some comments, we'll also need the new class for checking if an account is blocked.

Done. 
Implemented types to the routes in all endpoints.
Also, Implemented the ModerationChecker class in api/moderation.ts to verify accounts against global and domain-specific blocklists and allowlists.

RangerMauve commented on api/index.ts:
```diff
@@ -117,18 +117,120 @@ const v1Routes = (cfg: APIConfig, store: Store) => async (server: FastifyTypebox
+        200: Type.String()
+      },
+      description: 'Get global list of blocked users/instances as newline delimited string.',
+      tags: ['ActivityPub']
```
```suggestion
      tags: ['Moderation']
```

Since this isn't part of the activitypub spec, mind changing the tags to "moderation" so they'll show up in a seperate section?

RangerMauve reviewed (changes requested):
Nice, minor changes and else it looks perfect.
